### DISCUSSION TITLE
Which verifyOtp() type should be used for reauthenticate()?

### DISCUSSION THREAD

**Discussion by OliverBurrell** [supabase/supabase#31124] [Questions]
Trying to make the user account management page more secure. I am calling the reauthenticate() method before a user can change the account data. I am collecting the Otp in the same form as the account data changes and call the verifyOtp() method before the updateUser() method. If I use type 'email' or 'email_change' the verifyOtp() responds with 'Token has expired or is invalid'. Which type do I need to use or is this flow not supported?

**Comment by GaryAustin1** [comment]
The docs for reauthenticate specifically say the nonce is to be used with updateUser.  updateUser has an option to provide the nonce.
I have not used it though so am going by the docs:
![image](https://github.com/user-attachments/assets/12aa4b59-d9f9-435a-b100-098ae962fee0)

**Reply by OliverBurrell** [reply]
Yes, thanks, I have just read it as well...
Are there any methods I could use to send the user a nonce before they can change their account details?
The reason is security if a user leaves their desktop logged in and someone could change, e.g. their name.

**Reply by GaryAustin1** [reply]
Not something I've messed with.
You could have a built in signout and signin maybe?  The signout might impact other flow though.  Also could be bypassed if in client code.

**Reply by OliverBurrell** [reply]
Yes, I might try that - thanks.
